mlflow/pyfunc/__init__.py

```diff
@@ -2539,14 +2540,16 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                     if isinstance(each_message, ChatMessage):
                         messages.append(each_message)
                     else:
-                        messages.append(ChatMessage(**each_message))
+                        messages.append(ChatMessage.from_dict(each_message))
             else:
                 # If the input example is a dictionary, convert it to ChatMessage format
                 messages = [
-                    ChatMessage(**m) if isinstance(m, dict) else m
+                    ChatMessage.from_dict(m) if isinstance(m, dict) else m
                     for m in input_example["messages"]
                 ]
-                params = ChatParams(**{k: v for k, v in input_example.items() if k != "messages"})
+                params = ChatParams.from_dict(
+                    {k: v for k, v in input_example.items() if k != "messages"}
```

Comment:
We don't need this any longer due to from_dict() ignoring undefined arguments, right?

Reply:
oh yes this is true